### PR TITLE
fix: shared Rsbuild plugins should run only once

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

I found this issue when testing [rsbuild-plugin-publint](https://github.com/rspack-contrib/rsbuild-plugin-publint).

In Rsbuild, shared plugins will run only once. Take `rsbuild-plugin-publint` as an example, it uses `api.onAfterBuild` to trigger the linting, and this should only be triggered once per build, otherwise it will result in duplicate lint output.

```ts
// rsbuild.config.ts
export default {
  // this is a shared plugin
  plugins: [publintPlugin],
  environments: {
    client: {
      plugins: [clientOnlyPlugin],
    },
    server: {
      plugins: [serverOnlyPlugin],
    },
  },
};
```

Currently Rslib does not align with this behavior, the shared plugins are registered multiple times. After this PR, Rslib will be aligned with Rsbuild.

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/cli/inspect.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export async function inspect(
   const environments = await composeRsbuildEnvironments(config);
   const rsbuildInstance = await createRsbuild({
     rsbuildConfig: {
+      plugins: config.plugins,
```
Comment (Timeless0911) [inline packages/core/src/cli/inspect.ts:13]: missing `mf dev` commands in `packages/core/src/cli/mf.ts`?

Comment (chenjiahan) [reply packages/core/src/cli/inspect.ts:13]: Nice catch, this has been fixed

File: packages/core/src/cli/mf.ts [hunk 2]
```diff
@@ -25,8 +25,15 @@ async function initMFRsbuild(
...
 
   mfRsbuildConfig.config = changeEnvToDev(mfRsbuildConfig.config);
+
   const rsbuildInstance = await createRsbuild({
-    rsbuildConfig: mfRsbuildConfig.config,
+    rsbuildConfig: {
+      ...mfRsbuildConfig.config,
+      plugins: [
+        ...(rslibConfig.plugins || []),
+        ...(mfRsbuildConfig.config.plugins || []),
```
Comment (Timeless0911) [inline packages/core/src/cli/mf.ts:34]: Why `mfRsbuildConfig.config.plugins` is added to shared plugins?

Comment (chenjiahan) [reply packages/core/src/cli/mf.ts:34]: This is not the shared plugins, it's the merged final plugins

Comment (Timeless0911) [reply packages/core/src/cli/mf.ts:34]: Got it, we should refactor this to using `environments` as same as `build` and `inspect` command.

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**